backend/api/contacts/src/main/java/co/airy/core/contacts/dto/Contact.java

```diff
@@ -0,0 +1,225 @@
+                    .state(map.getMetadataValue(MetadataKeys.Address.STATE))
+                    .postalCode(map.getMetadataValue(MetadataKeys.Address.POSTAL_CODE))
+                    .country(map.getMetadataValue(MetadataKeys.Address.COUNTRY))
+                    .build();
+        }
+        @JsonIgnore
+        public List<Metadata> toMetadata(String contactId) {
+            List<Metadata> metadata = new ArrayList<>();
+            if (organizationName != null) {
+                metadata.add(newContactMetadata(contactId, MetadataKeys.Address.ORGANIZATION_NAME, organizationName));
+            }
+            if (addressLine1 != null) {
+                metadata.add(newContactMetadata(contactId, MetadataKeys.Address.ADDRESS_LINE1, addressLine1));
+            }
+            if (addressLine2 != null) {
+                metadata.add(newContactMetadata(contactId, MetadataKeys.Address.ADDRESS_LINE2, addressLine2));
+            }
+            if (city != null) {
+                metadata.add(newContactMetadata(contactId, MetadataKeys.Address.CITY, city));
+            }
+            if (state != null) {
+                metadata.add(newContactMetadata(contactId, MetadataKeys.Address.STATE, state));
+            }
+            if (postalCode != null) {
+                metadata.add(newContactMetadata(contactId, MetadataKeys.Address.POSTAL_CODE, postalCode));
+            }
+            if (country != null) {
+                metadata.add(newContactMetadata(contactId, MetadataKeys.Address.COUNTRY, country));
+            }
+            return metadata;
+        }
+    }
+
+    private Long createdAt;
+    private Long updatedAt;
+
+    public static class MetadataKeys {
+        public static String ID = "id";
+        public static String CREATED_AT = "created_at";
+        public static String DISPLAY_NAME = "displayName";
+        public static String AVATAR_URL = "avatarUrl";
+        public static String TITLE = "title";
+        public static String GENDER = "gender";
+        public static String TIMEZONE = "timezone";
+        public static String LOCALE = "locale";
+        public static String ORGANIZATION_NAME = "organizationName";
+        public static String VIA = "via";
+        public static String CONVERSATIONS = "conversations";
+        public static String METADATA = "metadata";
+
+        public static String ADDRESS = "address";
+
+        public static class Address {
+            public static String ORGANIZATION_NAME = "address.organizationName";
+            public static String ADDRESS_LINE1 = "address.address_line1";
+            public static String ADDRESS_LINE2 = "address.address_line2";
+            public static String POSTAL_CODE = "address.postal_code";
+            public static String CITY = "address.city";
+            public static String STATE = "address.state";
+            public static String COUNTRY = "address.country";
+        }
+    }
+
+
+    @JsonIgnore
+    public List<Metadata> toMetadata() {
```

Comment:
The contact object is stored entirely in metadata. Therefore we need a full mapping from one representation to the other.